src/ControllerV2/RosalieInteroperabilityController.php

```diff
@@ -18,7 +20,25 @@ class RosalieInteroperabilityController extends AbstractController
 {
     #[IsGranted('ROLE_MEMBRE')]
     #[Route('/beneficiaries/{id}/add-si-siao-number', name: 'add_si_siao_number')]
-    public function addSiSiaoNumber(Request $request, Beneficiaire $beneficiary, RosalieService $service, TranslatorInterface $translator): Response
+    public function addSiSiaoNumber(Request $request, Beneficiaire $beneficiary, RosalieService $service, TranslatorInterface $translator, EntityManagerInterface $em): Response
+    {
+        $beneficiaryCreationProcess = $beneficiary->getCreationProcess();
+
+        $redirection = $beneficiaryCreationProcess?->getIsCreating()
+            ? $this->redirectToRoute('create_beneficiary', ['id' => $beneficiaryCreationProcess->getId(), 'step' => $beneficiaryCreationProcess->getLastReachedStep()])
+            : $this->redirectToRoute('list_beneficiaries');
+
+        return $this->addSiSiaoNumberForm($request, $beneficiary, $service, $translator, $em, $redirection);
+    }
+
+    #[IsGranted('ROLE_MEMBRE')]
+    #[Route('/beneficiary/{id}/affiliate/add-si-siao-number', name: 'affiliate_beneficiary_add_si_siao_number')]
+    public function affiliationAddSiSiaoNumber(Request $request, Beneficiaire $beneficiary, RosalieService $service, TranslatorInterface $translator, EntityManagerInterface $em): Response
+    {
+        return $this->addSiSiaoNumberForm($request, $beneficiary, $service, $translator, $em, $this->redirectToRoute('affiliate_beneficiary_relays', ['id' => $beneficiary->getId()]));
+    }
+
```

Comment:
I don't know if addSiSiaoNumberForm is a correct method name, for a method that create a form instance, process form, and return a response